rom/laghos_rom.cpp

```diff
@@ -1091,6 +1103,44 @@ void ROM_Sampler::Finalize(Array<int> &cutoff, ROM_Options& input)
         delete tBasisE;
     }
 
+	if (input.hyperreductionSamplingType == eqp_energy)
+	{
+		const CAROM::Matrix *basisV = generator_V->getSpatialBasis();
+		const CAROM::Matrix *basisE = generator_E->getSpatialBasis();
+
+		MPI_Bcast(cutoff.GetData(), cutoff.Size(), MPI_INT, 0, MPI_COMM_WORLD);
+
+		// Truncate the bases.
+		// For the energy basis, the cutoff parameter has already been
+		// increased by 1 to accomodate the energy identity.
+		CAROM::Matrix *tBasisV = basisV->getFirstNColumns(cutoff[1]);
+		CAROM::Matrix *tBasisE = basisE->getFirstNColumns(cutoff[2]);
+
+		// Form the energy identity and replace the last basis vector by it. 
+		Vector unitE(input.tL2Size);
+
+		// TODO: my understanding is that the "=" opetaror is overloaded, so
+		// that, by setting unitE = 1.0, the vector unitE contains the coeffs
+		// that result in the summed function being 1.0 on the grid points. 
+		// Is that correct?
+		unitE = 1.0;
```

Comment:
Note my question here: does that lead to the correct energy identity vector? 

I saw that this is the way the energy identity is also computed in a different part of the code; I just want to make sure this is the right way, and the right vector size.